### DISCUSSION TITLE
How to proxy Cloudflare geo location headers to external API?

### DISCUSSION THREAD

**Discussion by huongphamx** [nuxt/nuxt#24890] [Questions] [answered]
Hi. I want to proxy Cloudflare geo location headers to my API (not Nuxt server). I used useRequestHeaders as in the docs, but headers are not send to my API. How can I get all headers from Cloudflare (as can get from cf Pages function logs) and proxy them?
```json
      "headers": {
        "accept": "text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,*/*;q=0.8",
        "accept-encoding": "gzip",
        "accept-language": "en-US,en;q=0.5",
        "cf-connecting-ip": "",
        "cf-connecting-o2o": "1",
        "cf-ipcity": "",
        "cf-ipcontinent": "",
        "cf-ipcountry": "",
        "cf-iplatitude": "",
        "cf-iplongitude": "",
        "cf-ray": "83b71c277bf15158",
        "cf-region": "",
        "cf-region-code": "",
        "cf-timezone": "",
        "cf-visitor": "{\"scheme\":\"https\"}",
        "connection": "Keep-Alive",
        "cookie": "REDACTED",
        "host": "",
        "sec-fetch-dest": "document",
        "sec-fetch-mode": "navigate",
        "sec-fetch-site": "none",
        "sec-fetch-user": "?1",
        "upgrade-insecure-requests": "1",
        "user-agent": "",
        "x-forwarded-for": "",
        "x-forwarded-proto": "https",
        "x-real-ip": ""
      },
```

**Comment by TheAlexLichter** [comment]
Can you share the current code? What did you try so far?

**Reply by huongphamx** [reply, marked as answer]
Hi, I think I solved parts of my problem by writing an API in server/ directory - that will again call my external API while passing headers from `headers: event.node.req.headers` - frontend will call that instead of calling directly my external API.
However, I think it's missing cf_ipcountry atm, I'll check it

**Reply by huongphamx** [reply]
For more info about my first approach: I firstly use a custom useFetch from [this example](https://nuxt.com/docs/examples/advanced/use-custom-fetch-composable) and [pass headers to the api](https://nuxt.com/docs/getting-started/data-fetching#pass-client-headers-to-the-api), but what I receive in external (python) API are none of headers above, look like they are from the CF CDN server.
